### PR TITLE
ps3-disc-dumper: init at 3.2.3

### PR DESCRIPTION
**Pull request by evanjs**
###### Description of changes
Initialize [ps3-disc-dumper](https://github.com/13xforever/ps3-disc-dumper) at [v3.2.3](https://github.com/13xforever/ps3-disc-dumper/releases/tag/v3.2.3)

## ~~Blocker~~
~~Depends on #182273, the package expects to be self-contained (see relevant snippet from pipeline [here](https://github.com/13xforever/ps3-disc-dumper/blob/99560ee3612c2bcfc128c0441dcf00e24aae2b95/azure-pipelines.yml#L15))~~
**Update: #182273 has been merged**

###### Things done

Tested using Ninja Gaiden Sigma
Game dumps properly using this utility and runs well on RCPS3

Note: Currently, IRD keys must be retrieved manually and placed in `~/.config/ps3-iso-dumper/ird/`
See https://github.com/13xforever/ps3-disc-dumper/issues/16#issuecomment-917484667 for more information
As the official release builds are not used, the redump snapshots are not included with the build output, and the IRD files must be provided outside the context of the build.
___

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by mdarocha — APPROVED** [review]

File: pkgs/tools/games/ps3-disc-dumper/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "--runtime linux-x64"
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/13xforever/ps3-disc-dumper";
+    description = "A handy utility to make decrypted PS3 disc dumps";
+    license = licenses.mit;
+    maintainers = with maintainers; [ evanjs ];
+    platforms = with platforms; linux;
+    mainProgram = "ps3-disc-dumper";
```
Comment (SuperSandro2000) [inline pkgs/tools/games/ps3-disc-dumper/default.nix:40]:
```suggestion
```

that should be the default

**Comment by mdarocha** [comment]
These 3 commits should probably be merged into one

**Comment by SuperSandro2000** [comment]
```
/nix/store/yshxqvl9g9vw497rb6qpxckfj06xdg3g-dotnet-sdk-6.0.301/sdk/6.0.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1114,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/build/source/IrdLibraryClient/IrdLibraryClient.csproj]
```

Interesting 🤔 

didn't know that

**Comment by evanjs** [comment]
> ```
> /nix/store/yshxqvl9g9vw497rb6qpxckfj06xdg3g-dotnet-sdk-6.0.301/sdk/6.0.301/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1114,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/build/source/IrdLibraryClient/IrdLibraryClient.csproj]
> ```
> 
> Interesting 🤔
> 
> didn't know that

Yep, seems to go away with #182273 when using `selfContainedBuild = true` 😄

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/tools/games/ps3-disc-dumper/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  src = fetchFromGitHub {
+    owner = "13xforever";
+    repo = "ps3-disc-dumper";
+    rev = "v${version}";
+    sha256 = "sha256-m3TS9H6cbEAHn6PvYQDMzdKdnOnDSM4lxCTdHBCXLV4=";
+  };
+
+  selfContainedBuild = true;
+
+  projectFile = [ "UI.Console/UI.Console.csproj" ];
```
Comment (IvarWithoutBones) [inline pkgs/tools/games/ps3-disc-dumper/default.nix:22]: This doesn't need to be a list:
```suggestion
  projectFile = "UI.Console/UI.Console.csproj";
```

File: pkgs/tools/games/ps3-disc-dumper/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  dotnetFlags = [
+    "--runtime linux-x64"
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/13xforever/ps3-disc-dumper";
+    description = "A handy utility to make decrypted PS3 disc dumps";
+    license = licenses.mit;
+    maintainers = with maintainers; [ evanjs ];
+    platforms = with platforms; linux;
```
Comment (IvarWithoutBones) [inline pkgs/tools/games/ps3-disc-dumper/default.nix:39]: Seeing how `--runtime linux-x64` is set, I think this should be enabled for `x86_64-linux`.

**Pull request merged**